src/backend/dataDirectory.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import path from "path";
+import fs from "fs-extra";
+import Database from "better-sqlite3";
+import stringify from "json-stable-stringify";
+import {type Project, projectToJSON} from "../core/project";
+import {directoryForProjectId} from "../core/project_io";
+import {type CacheProvider} from "./cache";
+import type {ProjectStorageProvider, ProjectStorageExtras} from "./project";
+
+/**
+ * Represents a SourceCred data directory.
+ */
+export class DataDirectory implements CacheProvider, ProjectStorageProvider {
+  sourcecredDirectory: string;
+  cacheDirectory: string;
+
+  constructor(sourcecredDirectory: string) {
+    this.sourcecredDirectory = sourcecredDirectory;
+    this.cacheDirectory = path.join(sourcecredDirectory, "cache");
```

Comment:
The `cacheDirectory` is a public property, however there doesn't seem to be any way for a client to know if the cacheDirectory has actually been created. So, to be safe, every client needs to do `await fs.mkdirp(dataDirectory.cacheDirectory)`. Seems messy. Can we give the `DataDirectory` and `initialize()` method, or just have it ensure directories exist in the constructor?

Reply:
It should probably be private. Since you're not supposed to do really any IO there when this class is responsible for that :]

Reply:
Made both private and immutable.
(Wasn't aware underscore properties were an actually supported Flow convention)